src/include/OSL/batched_texture.h

```diff
@@ -0,0 +1,278 @@
+                  == VecReg<16>::alignment,
+              "Expect alignment of data member to set alignment of struct");
+static_assert(std::alignment_of<VaryingTextureOptions<8>>::value
+                  == VecReg<8>::alignment,
+              "Expect alignment of data member to set alignment of struct");
+
+template<int WidthT> struct BatchedTextureOptions {
+    VaryingTextureOptions<WidthT> varying;
+    UniformTextureOptions uniform;
+
+    // Options set INTERNALLY by libtexture after the options are passed
+    // by the user.  Users should not attempt to alter these!
+    int private_envlayout = 0;  // Layout for environment wrap
+
+    // Implementation detail
+    // keep order synchronized to the data members in this structure
+    enum class LLVMMemberIndex {
+        sblur = 0,
+        tblur,
+        rblur,
+        swidth,
+        twidth,
+        rwidth,
+        firstchannel,
+        subimage,
+        subimagename,
+        swrap,
+        twrap,
+        rwrap,
+        mipmode,
+        interpmode,
+        anisotropic,
+        conservative_filter,
+        fill,
+        missingcolor,
+        private_envlayout,
+        count
+    };
+};
+static_assert(std::alignment_of<BatchedTextureOptions<16>>::value
+                  == VecReg<16>::alignment,
+              "Expect alignment of data member to set alignment of struct");
+static_assert(std::alignment_of<BatchedTextureOptions<8>>::value
+                  == VecReg<8>::alignment,
+              "Expect alignment of data member to set alignment of struct");
+
+#ifdef OIIO_TEXTURE_SIMD_BATCH_WIDTH
+// Code here is to validate our OSL BatchedTextureOptions<WidthT> is binary compatible
+// and safe to reinterpret_cast<TextureOptBatch*>
+static_assert((OIIO::Tex::BatchWidth == 16) || (OIIO::Tex::BatchWidth == 8),
+              "This validation requires OIIO_TEXTURE_SIMD_BATCH_WIDTH=16");
+
+namespace validate_offsets {
+
+OSL_PRAGMA_WARNING_PUSH
+OSL_GCC_PRAGMA(GCC diagnostic ignored "-Winvalid-offsetof")
+typedef BatchedTextureOptions<OIIO::Tex::BatchWidth> BTO;
+static constexpr size_t uniform_offset = offsetof(BTO, uniform);
+static constexpr size_t varying_offset = offsetof(BTO, varying);
+
+typedef VaryingTextureOptions<OIIO::Tex::BatchWidth> VTO;
+
+static_assert(offsetof(OIIO::TextureOptBatch, sblur) % 64 == 0,
+              "oops unaligned wide variable");
+static_assert(offsetof(OIIO::TextureOptBatch, tblur) % 64 == 0,
+              "oops unaligned wide variable");
+static_assert(offsetof(OIIO::TextureOptBatch, rblur) % 64 == 0,
+              "oops unaligned wide variable");
+static_assert(offsetof(OIIO::TextureOptBatch, swidth) % 64 == 0,
+              "oops unaligned wide variable");
+static_assert(offsetof(OIIO::TextureOptBatch, twidth) % 64 == 0,
+              "oops unaligned wide variable");
+static_assert(offsetof(OIIO::TextureOptBatch, rwidth) % 64 == 0,
+              "oops unaligned wide variable");
```

Comment:
I don't even want to change it in this PR, but just food for future thought: Would it be better to tuck away these static checks in just one .cpp module (any one will do, right?) rather than have them parsed and evaluated for every translation unit that includes the header?

Reply:
Depends on one's confidence level, they could be moved into some type of unit test that builds and enforces this.  But that moves intent further away from enforcement.  Do we think there is a compile time cost (measurable, I'm sure its not free).  Honestly the motivation for moving it out would just be so reader doesn't have to see it there.